ui/src/app/utils/access-utils.spec.tsx

```diff
@@ -1,118 +1,243 @@
+        }]
+      }
+    }
+
     expect(maybeDaysRemaining(laterExpiration)).toBeUndefined();
   });
 
   it('returns the soonest of all expirations within the window', () => {
-    expect(maybeDaysRemaining(expirationsInWindow)).toEqual(5);
+    const soonDays = 5;
+    const aBitLater = 10;
+
+    const expirationsInWindow: Profile = {
+      ...ProfileStubVariables.PROFILE_STUB,
+      accessModules : {
+        modules: [{
+          moduleName: AccessModule.COMPLIANCETRAINING,
+          expirationEpochMillis: nowPlusDays(soonDays) + ONE_MINUTE_IN_MILLIS,
+        }, {
+          moduleName: AccessModule.DATAUSERCODEOFCONDUCT,
+          expirationEpochMillis: nowPlusDays(aBitLater) + ONE_MINUTE_IN_MILLIS,
+        }]
+      }
+    }
+
+    expect(maybeDaysRemaining(expirationsInWindow)).toEqual(soonDays);
   });
 
   // regression test for RW-7108
   it('returns 30 days when the max expiration is between 30 and 31 days', () => {
+    const thirtyDaysPlusExpiration: Profile = {
+      ...ProfileStubVariables.PROFILE_STUB,
+      accessModules : {
+        modules: [{
+          moduleName: AccessModule.COMPLIANCETRAINING,
+          expirationEpochMillis: nowPlusDays(30) + ONE_MINUTE_IN_MILLIS,
+        }, {
+          moduleName: AccessModule.DATAUSERCODEOFCONDUCT,
+          expirationEpochMillis: nowPlusDays(31) + ONE_MINUTE_IN_MILLIS,
+        }]
+      }
+    }
+
     expect(maybeDaysRemaining(thirtyDaysPlusExpiration)).toEqual(30);
   });
 });
 
+describe('computeRenewalDisplayDates', () => {
+  const EXPIRATION_DAYS = 123;  // arbitrary for testing; actual prod value is 365
+  const LOOKBACK_PERIOD = 99;   // arbitrary for testing; actual prod value is 330
+
+  beforeEach(() => {
+    serverConfigStore.set({
+      config: {
+        ...defaultServerConfig,
+        accessRenewalLookback: LOOKBACK_PERIOD
+      }
+    });
+  });
+
+  it('returns Unavailable/Incomplete when the module is incomplete', () => {
+    const expected = {
+      lastConfirmedDate: 'Unavailable (not completed)',
+      nextReviewDate: 'Unavailable (not completed)'
+    };
+    expect(computeRenewalDisplayDates({})).toStrictEqual(expected);
+  });
+
+  it('returns Unavailable/Incomplete when the module is incomplete, regardless of expiration date', () => {
+    const expirationDate = nowPlusDays(25);
+    const status: AccessModuleStatus = {
+      expirationEpochMillis: expirationDate
+    };
+    const expected = {
+      lastConfirmedDate: 'Unavailable (not completed)',
+      nextReviewDate: 'Unavailable (not completed)'
+    };
+    expect(computeRenewalDisplayDates(status)).toStrictEqual(expected);
+  });
+
+  it('returns Unavailable/Bypassed when the module is bypassed, regardless of bypass date', () => {
+    const bypassDate = nowPlusDays(-10000);
+    const status: AccessModuleStatus = {
+      bypassEpochMillis: bypassDate
+    };
+    const expected = {
```

Comment:
opt nit: I might inline these rather than having an `expected` variable, but that's personal preference.

If you keep it this way - these are all missing `;`s